### ISSUE TITLE
Error in chartist.bar

### ISSUE THREAD

**Issue by willsoto** [chartist-js/chartist#69]
The following [line](https://github.com/gionkunz/chartist-js/blob/afe23c18a10f7befac7d3fba2abe63cf9e61b848/source/scripts/chartist.bar.js#L269) is throwing an `undefined is not a function` error.

When `$watch`ing data for changes, the [angular directive](https://github.com/paradox41/ng-chartist.js/blob/master/src/ng-chartist.js#L33-L39) calls `chart.detach()`. This is where the error is being thrown.

**Comment by gionkunz** [comment]
Hi Will, let me investigate on this.

**Comment by willsoto** [comment]
If you want a stack trace, let me know

**Comment by gionkunz** [comment]
Hey Will, I've just found the bug that caused these issues. I missed some stuff during the event refactoring. 0.2.3 is released now with just this bug fix.

**Comment by willsoto** [comment]
:+1:
